src/main/java/org/folio/mr/service/impl/MediatedRequestsServiceImpl.java

```diff
@@ -76,4 +97,87 @@ public Optional<MediatedRequest> delete(UUID requestId) {
         return requestsMapper.mapEntityToDto(mediatedRequestEntity);
       });
    }
+
+  private MediatedRequest setStatusBasedOnMediatedRequestStatusAndStep(
+    MediatedRequest mediatedRequest) {
+
+    var statusElements = mediatedRequest.getStatus().toString().split(" - ");
+    if (statusElements.length == 2) {
+      mediatedRequest.setMediatedRequestStatus(
+        MediatedRequest.MediatedRequestStatusEnum.fromValue(statusElements[0]));
+      mediatedRequest.setMediatedRequestStep(statusElements[1]);
```

Comment:
So step is just a string? Why not another enum?

Reply:
Different workflows have different steps